src/Plan/Plan.php

```diff
@@ -111,6 +119,25 @@ public function setName(string $name)
         return $this;
     }
 
+    /**
+     * @return array
+     */
+    public function paymentMethod()
+    {
+        return $this->paymentMethod ?? $this->firstPaymentMethod;
+    }
+
+    /**
+     * @param  array  $paymentMethod
+     * @return $this
+     */
+    public function setMethod($paymentMethod)
```

Comment:
Naming consistency: either use `payment method` or `method`.

Reply:
BTW, I prefer the _plural_ `first payment methods` for verbosity. But refactoring everything into that (except for the Mollie payload) would require a separate PR.

Reply:
In the `ConfigPlanRepository` -> `populatePlan` you have

 ```php 
default: // call $plan->setKey() if setKey method exists
    $method = 'set' . ucfirst($key);
    if (method_exists($plan, $method)) {
        $plan->$method($value);
    }

break;
``` 
and the config file uses `method` to define allowed payment methods. I use `setMethod($paymentMethod)` to be clear about what method is about